compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/alias-capture-in-method-receiver-and-mutate.expect.md

```diff
@@ -34,13 +34,12 @@ function Component() {
   const $ = _c(1);
   let t0;
   if ($[0] === Symbol.for("react.memo_cache_sentinel")) {
+    const x = [];
     const a = makeObject_Primitives();
 
-    const x = [];
+    t0 = [x, a];
     x.push(a);
-
     mutate(x);
```

Comment:
this gives a pretty good intuition for the algorithm:
* the return value is the `[x, a]` array, so we start emitting from there.
* we emit in dependency order, so we emit the `x` instruction first. That emits all the serialized reads/writes of x and their dependencies. In this case there is the `const x = []` and then a few references to x, so we actually emit a bunch of `LoadLocal x` right after the `const x = []`, but those shift later during codegen to where they're used.
* then we emit `a`, and its dependency on the `makeObject_Primitives()` call instr.
* now we're done emitting what we "need" to satisfy the terminal's dependencies, and we emit everything remaining starting from later instructions first.
* this means we start from the `mutate(x)` instruction. it has an implicit ordering dependency on `x.push(a)`, so we emit that dependency first so that we keep the `x.push(a); mutate(x)` in-order.